### PR TITLE
Add a new page tracking Deprecation of features in MLIR

### PR DESCRIPTION
**Pull request by joker-eph**

### CODE REVIEW

File: website/content/deprecation/_index.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+title: "Depreciations"
+date: 2023-02-16T15:26:15Z
+draft: false
+weight: 2
+---
+
+This page collects current depreciation in MLIR of API and features we intend
+to remove soon.
+
+## Use free function for `dyn_cast`/`cast`
```
Comment (kuhar) [inline website/content/deprecation/_index.md:11]: What do you think about also including major refactorings, e.g., the switch to the prefixed accessor API, and linking to the relevant discourse thread?

I'm thinking about splitting this into 3 sections:
1. Upcoming refactorings / deprecations.
2. Ongoing refactorings / deprecations.
3. Past refactorings / deprecations.

File: website/content/deprecation/_index.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+When casting attributes or type, use the free functions variants, e.g.,
+`dyn_cast<T>(x)`, `isa<T>(x)`, etc. Use of the cast methods variants (e.g.,
+`x.dyn_cast<T>()`) should be avoided in new code as we'll remove these
+methods in the future.
```
Comment (kuhar) [inline website/content/deprecation/_index.md:16]: Would it help to link to the relevant discourse thread?

File: website/content/deprecation/_index.md (hunk 1 above)
Comment (kuhar) [inline website/content/deprecation/_index.md:11]: nit: `dyn_cast`/`cast`/`isa`/etc, since there are more cast functions than those two

Comment (joker-eph) [reply website/content/deprecation/_index.md:11]: Yeah these seems like nice additions!

File: website/content/deprecation/_index.md (hunk 2 above)
Comment (joker-eph) [reply website/content/deprecation/_index.md:16]: Thanks, good idea!

**Comment by bondhugula** [comment]
Spell typo in PR/commit title: "Deprecation"

**Comment by joker-eph** [comment]
> Spell typo in PR/commit title: "Deprecation"

Pardon my French ;)

**Comment by joker-eph** [comment]
I'll land this, let's iterate post-landing! Feel free to still comment if you have thoughts.

**Pull request merged**